tests/scripts/depends.py

```diff
@@ -258,15 +260,18 @@ def test(self, options):
     'MBEDTLS_DES_C': ['PSA_WANT_KEY_TYPE_DES'],
     'MBEDTLS_GCM_C': ['PSA_WANT_ALG_GCM'],
 
-    'MBEDTLS_CIPHER_MODE_CBC': ['PSA_WANT_ALG_CBC_PKCS7',
-                                'PSA_WANT_ALG_CBC_NO_PADDING'],
-    'MBEDTLS_CIPHER_MODE_CFB': ['PSA_WANT_ALG_CFB'],
-    'MBEDTLS_CIPHER_MODE_CTR': ['PSA_WANT_ALG_CTR'],
-    'MBEDTLS_CIPHER_MODE_OFB': ['PSA_WANT_ALG_OFB'],
+    'PSA_WANT_ALG_CBC_NO_PADDING': ['MBEDTLS_CIPHER_MODE_CBC'],
+    'PSA_WANT_ALG_CBC_PKCS7': ['MBEDTLS_CIPHER_MODE_CBC'],
+    'PSA_WANT_ALG_CFB': ['MBEDTLS_CIPHER_MODE_CFB'],
+    'PSA_WANT_ALG_CTR': ['MBEDTLS_CIPHER_MODE_CTR'],
+    'PSA_WANT_ALG_OFB': ['MBEDTLS_CIPHER_MODE_OFB'],
 
     'MBEDTLS_CIPHER_PADDING_PKCS7': ['MBEDTLS_PKCS5_C',
                                      'MBEDTLS_PKCS12_C',
                                      'PSA_WANT_ALG_CBC_PKCS7'],
+    'MBEDTLS_CIPHER_PADDING_ONE_AND_ZEROS': ['MBEDTLS_CIPHER_MODE_CBC'],
+    'MBEDTLS_CIPHER_PADDING_ZEROS': ['MBEDTLS_CIPHER_MODE_CBC'],
+    'MBEDTLS_CIPHER_PADDING_ZEROS_AND_LEN': ['MBEDTLS_CIPHER_MODE_CBC'],
```

Comment:
To test the padding mode in an exclusive group domain only one must be enabled in the group. To disable `MBEDTLS_CIPHER_PADDING_PKCS7`, `PSA_WANT_ALG_CBC_PKCS7` also need to be disabled which will disable `MBEDTLS_CIPHER_MODE_CBC`. All padding mode needs `MBEDTLS_CIPHER_MODE_CBC` so to prevent it to be disabled, add it to the dependencies of the other padding modes then it will be in the excluded list.

Reply:
Thanks for the explanation. I forgot about the cipher_padding group actually. The way we handle the legacy configuration options is probably not optimal in cipher_chaining and cipher_padding domains but we are going to remove these configuration options soon thus no need to try and improve things.